src/air/public_memory.cairo

```diff
@@ -43,6 +43,48 @@ impl PageImpl of PageTrait {
             i += 1;
         }
     }
+
+    fn extract_range(self: @Page, addr: felt252, len: felt252) -> Span<felt252> {
+        let mut arr = ArrayTrait::new();
+        let mut i = 0;
+
+        loop {
+            if i == len {
+                break arr.span();
+            }
+
+            let current = *self.at((addr + i).try_into().unwrap());
```

Comment:
we can eliminate try into rangechecks by changing declaration of this func to take addr: u32 as an input

so logic is when u call it u ned to make try_into from felt 
big felts do not represent memory address because they are to large so it is logical to take addr: u32 to this func

WDYT?

```suggestion
            let current = *self.at(addr + i);
```

Reply:
I agree with that, but it implies that this type conversion will have to be done somewhere else, the reason for this was that everywhere this function is called the address is typically computed by doing some operations with `felt252`, thus you would have to convert it every single time before passing`

Reply:
That is right, but from opts point of view we should do this before passing like u said.
Otherwise rangecheck are going to be applied on every iteration of the loop.
Up to u tbh